syntax = "proto3";

service HelloWorld {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc StreamGreetings (HelloRequest) returns (stream HelloReply) {}
  rpc StreamClientGreetings (stream HelloRequest) returns (HelloReply);
  rpc bothStreamGreetings (stream HelloRequest) returns (stream HelloReply);
}

message HelloRequest {
  string name = 1;
  string language = 2; 
}

message HelloReply {
  string message = 1;
}
